dask/dataframe/io/demo.py

```diff
@@ -8,15 +13,139 @@
+
+default_int_args: dict[str, tuple[tuple[Any, ...], dict[str, Any]]] = {
+    "poisson": ((), {"lam": 1000}),
+    "normal": ((), {"scale": 1000}),
+    "uniform": ((), {"high": 1000}),
+    "binomial": ((1000, 0.5), {}),
+}
+
+
+@dataclass
+class ColumnSpec:
+    """Encapsulates properties of a family of columns with the same dtype.
+    Different method can be specified for integer dtype ("poisson", "uniform",
+    "binomial", etc.)
+
+    Notes
+    -----
+    This API is still experimental, and will likely change in the future"""
+
+    prefix: str | None = None
+    """Column prefix. If not specified, will default to str(dtype)"""
+
+    dtype: str | type | None = None
+    """Column data type"""
+
+    number: int = 1
+    """How many columns to create with these properties. Default 1.
+    If more than one columns are specified, they will be numbered: "int1", "int2", etc."""
+
+    nunique: int | None = None  # number of unique categories
+    """For a "category" column, how many unique categories to generate"""
+
+    choices: list = field(default_factory=list)
+    """For a "category" or str column, list of possible values"""
+
+    low: int | None = None
+    """Start value for an int column. Optional if random=True, since ``randint`` doesn't accept
+    high and low."""
+
+    high: int | None = None
+    """For a "category" column, how many unique categories to generate"""
+
+    length: int | None = None
+    """For a str or "category" column, how many unique categories to generate"""
+
+    random: bool = False
+    """For an int column, whether to use ``randint``"""
+
+    method: str | None = None
+    """For an int column, method to use when generating the value, such as "poisson", "uniform", "binomial".
+    Default "poisson". Delegates to the same method of ``RandomState``"""
+
+    kwargs: dict[str, Any] = field(default_factory=dict)
+    """For an int column, any other kwargs to pass into the method"""
+
+
+@dataclass
+class IndexSpec:
+    """Properties of the dataframe index
 
+    Notes
+    -----
+    This API is still experimental, and will likely change in the future"""
 
-def make_float(n, rstate):
+    dtype: str | type = int
+    """Index dtype. Currently only supports integer and datetime types"""
+
+    start: str | None = None
+    """First value of the index, only required for a DatetimeIndex"""
+
+    freq: int | str = 1
+    """Step for a RangeIndex, frequency for a DatetimeIndex ("1H", "1D", etc.)"""
+
+    partition_freq: str | None = None
+    """Partition frequency, required for a DatetimeIndex ("1D", "1M", etc.)"""
+
+
+@dataclass
+class DatasetSpec:
+    """Defines a dataset with random data, such as which columns and data types to generate
+
+    Notes
+    -----
+    This API is still experimental, and will likely change in the future"""
+
+    npartitions: int = 1
+    """How many partitions generate in the dataframe. If the dataframe has a DatetimeIndex, specify
```

Comment:
This is confusing, how does partition_freq impact this?

Reply:
`partition_freq` should be set for a dataframe with datetime index. This one is relevant for a dataframe with an integer index.

I'm not terribly interested in the datetime index support right now, because we already have `datasets.timeseries` for that, so the API for this use case is still a little clunky.